tags.go

```diff
@@ -204,56 +177,44 @@ func (conn *TagConnection) Hydrate(service ID, client *Client) error {
 	return nil
 }
 
+// Deprecated: use client.GetServiceWithAlias(alias).GetTags(client, nil) instead
 func (client *Client) GetTagsForServiceWithAlias(alias string) ([]Tag, error) {
-	service, serviceErr := client.GetServiceIdWithAlias(alias)
-	if serviceErr != nil {
-		return nil, serviceErr
+	s, err := client.GetServiceWithAlias(alias)
+	if err != nil {
+		return nil, err
 	}
-	return client.GetTagsForService(service.Id)
+	t, err := s.GetTags(client, nil)
+	return t.Nodes, err
 }
 
-// Deprecated: use GetTagsForService instead
+// Deprecated: use client.GetService(id).GetTags(client, nil) instead
 func (client *Client) GetTagsForServiceWithId(service ID) ([]Tag, error) {
-	return client.GetTagsForService(service)
+	s, err := client.GetService(service)
+	if err != nil {
+		return nil, err
+	}
+	t, err := s.GetTags(client, nil)
+	return t.Nodes, err
 }
 
+// Deprecated: use client.GetService(id).GetTags(client, nil) instead
 func (client *Client) GetTagsForService(service ID) ([]Tag, error) {
-	var q struct {
-		Account struct {
-			Service struct {
-				Tags TagConnection `graphql:"tags(after: $after, first: $first)"`
-			} `graphql:"service(id: $service)"`
-		}
-	}
-	v := PayloadVariables{
-		"service": service,
-		"after":   graphql.String(""),
-		"first":   client.pageSize,
-	}
-	if err := client.Query(&q, v, WithName("ServiceTagsList")); err != nil {
-		return q.Account.Service.Tags.Nodes, err
+	s, err := client.GetService(service)
+	if err != nil {
+		return nil, err
 	}
-	if err := q.Account.Service.Tags.Hydrate(service, client); err != nil {
-		return q.Account.Service.Tags.Nodes, err
-	}
-	return q.Account.Service.Tags.Nodes, nil
+	t, err := s.GetTags(client, nil)
+	return t.Nodes, err
 }
 
+// Deprecated: use client.GetService(id).GetTags(client, nil).TotalCount instead
```

Comment:
Do we have to make the GetTags call? For some reason I thought GetService would return tags too since it's supposed to run the hydrate function.

Reply:
DOH! - you are correct!  I will fix this